catroid/src/main/java/org/catrobat/catroid/utils/FormulaEditorIntroUtil.java

```diff
@@ -0,0 +1,261 @@
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_compute;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_object;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_functions;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_logic;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_device;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_data;
+		idx = 0;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_formula_editor;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_input_field;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_keyboard;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_compute;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_object;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_functions;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_logic;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_device;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_data;
+	}
+
+	private static void setParamsOfView(View view, int highlightRectangleIndex, int widthModifier, int heightModifier) {
+		int width = 1;
+		int height = 1;
+		int[] coords = new int[2];
+		coords[0] = 0;
+		coords[1] = 0;
+
+		if (view != null) {
+			width = view.getWidth() * widthModifier + (widthModifier - 1) * MODIFIER_ADDITION;
+			height = view.getHeight() * heightModifier + (widthModifier - 1) * MODIFIER_ADDITION;
+			view.getLocationOnScreen(coords);
+		}
+		highlightAreas[highlightRectangleIndex] = new Rectangle(coords[0], coords[1], width, height);
+	}
+
+	private static void setIntroTextPosition(View fragmentView) {
+		int[] coords = new int[2];
+		View view = (View) fragmentView.findViewById(R.id.formula_editor_edit_field);
+		view.getLocationOnScreen(coords);
+
+		TextView titleTextView = (TextView) introWindow.getContentView().findViewById(R.id.intro_title);
+		RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams) titleTextView.getLayoutParams();
+		params.topMargin = coords[1] + view.getHeight() + TEXT_SPACING;
+	}
+
+	private static void setScreenshotBitmaps(View fragmentView) {
+		SnackbarUtil.hideActiveSnack();
+
+		View screenShotView = fragmentView.getRootView();
+		screenShotView.setDrawingCacheEnabled(true);
+		screenShotView.layout(0, 0, screenShotView.getMeasuredWidth(), screenShotView.getMeasuredHeight());
+		screenshot = Bitmap.createBitmap(screenShotView.getMeasuredWidth(), screenShotView
+				.getMeasuredHeight(), Bitmap.Config.ARGB_8888);
+		Canvas canvas = new Canvas(screenshot);
+		screenShotView.draw(canvas);
+		screenShotView.setDrawingCacheEnabled(false);
+
+		screenshotDark = screenshot.copy(Bitmap.Config.ARGB_8888, true);
+		canvas = new Canvas(screenshotDark);
+		Paint paint = new Paint(Color.WHITE);
+		ColorFilter filter = new LightingColorFilter(0xFF5F5F5F, 0x00000000);
```

Comment:
those colors could also be either final static values at the top of the class, or values in the colors.xml